share/getters/shrex.go

```diff
@@ -25,6 +31,55 @@ const (
 	defaultMinAttemptsCount  = 3
 )
 
+var meter = global.MeterProvider().Meter("shrex/getter")
+
+type metrics struct {
+	edsAttempts syncint64.Histogram
+	ndAttempts  syncint64.Histogram
+}
+
+func (m *metrics) recordEDSAttempt(ctx context.Context, attemptCount int, success bool) {
```

Comment:
Looks like for each request with multiple attempts you want to observe sequence (1,2,3,4,5...). Maybe just observe final amount of attempts?

Reply:
No, we only observe the final amount - it only calls recordEDSAttempt before returns

Reply:
Thank you for clarification, now I got it